src/file_pipe_log/pipe.rs

```diff
@@ -22,10 +23,91 @@ use crate::{perf_context, Error, Result};
 use super::format::{FileNameExt, LogFileFormat};
 use super::log_file::{build_file_reader, build_file_writer, LogFileWriter};
 
+#[repr(u8)]
+#[derive(Clone, Copy, PartialEq, Eq, Debug, EnumIter)]
+pub enum StorageDirType {
+    Main = 0,
+    Secondary = 1,
+}
+
+/// Represents the info of storage dirs, including `main dir` and
+/// `secondary dir`.
+struct StorageInfo {
```

Comment:
StorageManager, and use path_id to refer to path instead of storage_type.

Reply:
path_id as in an integer. You can make the StorageManager manages a vector of directories, and you can also let it take care of directory lock as well. (Pass it to `Pipe::open`).